src/CloudNative.CloudEvents/CloudEventAttributes.cs

```diff
@@ -369,5 +372,14 @@ private bool ValidateAndNormalize(string key, ref object value)
 
             return false;
         }
+
+        private void ValidateKey(string key)
+        {
+            if (key.All(c => char.IsDigit(c) || char.IsLower(c)))
```

Comment:
I believe this would allow non-ASCII characters. I think it's simplest to use `(c >= 'a' && c <= 'z') || (c >= '0' && c <= '9')`